compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ASTBuilderUtil.java

```diff
@@ -868,12 +868,12 @@ public static BInvokableSymbol duplicateInvokableSymbol(BInvokableSymbol invokab
 
         if (Symbols.isFlagOn(invokableSymbol.flags, Flags.ISOLATED)) {
             dupFuncSymbol.flags |= Flags.ISOLATED;
-            dupInvokableType.flags |= Flags.ISOLATED;
+            dupInvokableType.setFlags(dupInvokableType.getFlags() | Flags.ISOLATED);
```

Comment:
Since the pattern `t.setFlags(t.getFlags() | newFlag);` seems to be somewhat common shall we introduce a new API `addFlags`  that internally access the flags field and call `setFlags`? Note sure if it will have significant effect, but should allow us to replace the `getFlags` function call with a field access (and `setFlags` should get inlined inside the `addFlags` method)

Reply:
Fixed